S/SuperLU_MT/build_tarballs.jl

```diff
@@ -0,0 +1,81 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "SuperLU_MT"
+version = v"3.1"
+
+# Collection of sources required to build Elemental
+sources = [
+    ArchiveSource("https://portal.nersc.gov/project/sparse/superlu/superlu_mt_3.1.tar.gz",
+                  "407b544b9a92b2ed536b1e713e80f986824cf3016657a4bfc2f3e7d2a76ecab6"),
+    DirectorySource("bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd SuperLU_MT_*
+
+cp MAKE_INC/make.linux.openmp make.inc
+
+atomic_patch -p1 ../patches/01-fix-makefile.patch
+
+# Build object files suitable for a shared library, and link the libraries we need.
+echo "CFLAGS += -fPIC -fopenmp \$(BLASLIB)" >> make.inc
+echo "NOOPTS += -fPIC -fopenmp \$(BLASLIB)" >> make.inc
+
+# On 64-bit systems, use 64-bit integers for indexing.
+if [[ "$nbits" == 64 ]]; then
```

Comment:
If we are building this for Sundials and modify Sundials to work with only 32-Bits 

"
Moreover, since the superlumt library may be installed to support either 32-bit or 64-bit integers, it is assumed that the superlumt library is installed using the same integer precision as the sundials sunindextype option
"

cc. @ViralBShah

Reply:
Yes, I think it makes sense to make these binaries 32-bit only to avoid the 32/64-bit mess. @chrisrackauckas fyi.

Reply:
So we're doing 32-bit everywhere, and so the `-D_LONGINT` should go away?

Reply:
Yes, let's do that.